### PR TITLE
Feat: Add CodeQL Analyzer

### LINKED ISSUE #4378: 🔎 Add Code Analyzer To ‘utopia-php/demo’
# 💭 Introduction

Code analyzer is one of the most valuable tools in the open source world. It not only ensures every high quality of the code but also allows CI/CD to ensure this. A code analyzer can find a potential bug, unexpected null value, or type mismatch. Having a code analyzer set up should be a high priority for every project as it makes the project more stable and bug-proof.

Your task is to update our library [utopia-php/demo](https://github.com/utopia-php/demo) and implement a code analyzer. We would like to use [PHPStan](https://phpstan.org/) as our analyzer.

**To make this task easy, we created a sample pull request showing what exactly needs to be done to implement the analyzer into the Utopia PHP library: [https://github.com/utopia-php/ab/pull/8](https://github.com/utopia-php/ab/pull/8)**

### 🎯 Requirements

- Experience with GitHub Actions.
- Experience with Composer.
- Experience with PHP.

### ✅ Task Summary

- [ ]  Ask to be assigned to the issue.
- [ ]  Wait to be assigned.
- [ ]  Implement code analyzer in [utopia-php/demo](https://github.com/utopia-php/demo) (use [https://github.com/utopia-php/ab/pull/8](https://github.com/utopia-php/ab/pull/8) as template)
- [ ]  Run code analyzer to check existing code, and update it if required
- [ ]  Submit a pull request in [utopia-php/demo](https://github.com/utopia-php/demo).

> Please configure PHPStan to max level (9). If that gives you too many problems that you cannot solve, please drop to level 8. If you have trouble with level 8 as well, let us know, and we can consider a lower level on some repositories.

If you have questions, need any help, or just want to hang out, make sure to join us on our [Discord server](https://appwrite.io/discord).

**Happy Appwriting!**

### PR DESCRIPTION
**Pull request by thinkverse**
This PR adds PHPStan to analyze the demo app for code quality. While not strictly needed in a demo app, it does show a user looking through the demo how to add it to their project. 👍

Closes https://github.com/appwrite/appwrite/issues/4378

### CODE REVIEW

**Comment by Meldiron** [comment]
@thinkverse , thanks for the PR! 🤯 Please give us some time to review it.

**Review by christyjacob4 — APPROVED** [review]
PR looks good 👍  Will wait for reviews from @Meldiron

File: .github/workflows/codeql-analysis.yml
Comment (christyjacob4) [inline .github/workflows/codeql-analysis.yml:20]:
```suggestion
        "composer install --profile --ignore-platform-reqs && composer check"
```

You're missing the double quotes which is preventing the check from running

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments

**Comment by thinkverse** [comment]
> Please address the comments

Done and done. 👍

**Comment by Meldiron** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Great PR! 🤯 Please make sure to:

1. update the composer.json to include the check script 
2. run the check and fix any problems

**Comment by thinkverse** [comment]
> 1. update the composer.json to include the check script

Great catch, seems when I merged in the latest from main (https://github.com/utopia-php/demo/pull/11/commits/a3023e3e7c8b936c574114854ce8360530e78ffd) the check script I had added before got removed. 😂

**Comment by thinkverse** [comment]
If someone could take care of the comment block above `App::init()` later since it doesn't match anymore that would be great. 👍 Appwrite team members will be able to re-write the comment with a proper explanation.

**Review by stnguyen90 — APPROVED** [review]
Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon!

**Comment by thinkverse** [comment]
OK, now this PR has been fixed of the conflict error caused by the latest merge into `main`, fixed some PHP version issues since the demo isn't locked to a specific PHP version. 👍

**Review by stnguyen90 — APPROVED** [review]
Hey 👋 awesome work on your PR! We've approved your work and it'll be merged soon!

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**
